### ISSUE TITLE
bug(list): Missing bloatwares (Xiaomi Poco X6 Pro)

### ISSUE THREAD

**Issue by cretoo** [Universal-Debloater-Alliance/universal-android-debloater-next-generation#592]
### Describe the bug

Some of the bloatwares are not listed in UAD, e.g. netflix, facebook, spotify music, the preinstalled game apps, etc.

### Expected behavior

Should be listed in UAD for uninstalling

### Operating System

Windows

### Provide logs

2024-08-25 00:18:49 DEBUG [src\core\update.rs:180] Checking for UAD-ng update
2024-08-25 00:18:49 INFO  [src\gui\views\list.rs:144] -----------------------------------------------------------------
2024-08-25 00:18:49 INFO  [src\gui\views\list.rs:145] ANDROID_SDK: 34 | DEVICE: POCO 2311DRK48G
2024-08-25 00:18:49 INFO  [src\gui\views\list.rs:149] -----------------------------------------------------------------
2024-08-25 00:20:13 INFO  [src\gui\views\list.rs:144] -----------------------------------------------------------------
2024-08-25 00:20:13 INFO  [src\gui\views\list.rs:145] ANDROID_SDK: 34 | DEVICE: POCO 2311DRK48G
2024-08-25 00:20:13 INFO  [src\gui\views\list.rs:149] -----------------------------------------------------------------
2024-08-25 00:21:24 ERROR [src\core\config.rs:79] Failed to read config file: `The system cannot find the file specified. (os error 2)`
2024-08-25 00:21:24 ERROR [src\core\config.rs:81] Restoring default config file
2024-08-25 00:21:24 DEBUG [src\core\update.rs:180] Checking for UAD-ng update
2024-08-25 00:21:24 INFO  [src\gui\views\list.rs:144] -----------------------------------------------------------------
2024-08-25 00:21:24 INFO  [src\gui\views\list.rs:145] ANDROID_SDK: 34 | DEVICE: POCO 2311DRK48G
2024-08-25 00:21:24 INFO  [src\gui\views\list.rs:149] -----------------------------------------------------------------
2024-08-25 00:25:47 DEBUG [src\core\update.rs:180] Checking for UAD-ng update
2024-08-25 00:25:47 DEBUG [src\core\update.rs:204] Failed to check UAD-ng update
2024-08-25 00:26:11 DEBUG [src\core\update.rs:180] Checking for UAD-ng update
2024-08-25 00:26:23 INFO  [src\gui\views\list.rs:144] -----------------------------------------------------------------
2024-08-25 00:26:23 INFO  [src\gui\views\list.rs:145] ANDROID_SDK: 34 | DEVICE: POCO 2311DRK48G
2024-08-25 00:26:23 INFO  [src\gui\views\list.rs:149] -----------------------------------------------------------------
2024-08-25 00:29:01 DEBUG [src\core\update.rs:180] Checking for UAD-ng update
2024-08-25 00:29:02 INFO  [src\gui\views\list.rs:144] -----------------------------------------------------------------
2024-08-25 00:29:02 INFO  [src\gui\views\list.rs:145] ANDROID_SDK: 34 | DEVICE: POCO 2311DRK48G
2024-08-25 00:29:02 INFO  [src\gui\views\list.rs:149] -----------------------------------------------------------------

### Acknowledgements

- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by cretoo** [comment]
Tried it on another phone (xiaomi poco m6 pro) and none of the bloatwares that i mentioned appeared too.

**Comment by Rudxain** [comment]
That's weird. I remember an issue where no packs would show up in the GUI. This seems related

**Comment by AnonymousWP** [comment]
We should pick this up if this really is a severe bug.

**Comment by cretoo** [comment]
Is there any chance of this being fixed?

**Comment by Rudxain** [comment]
@cretoo I guess most (all?) org members don't have multi-user devices, to test updates. So we'll have to redirect users (such as you) to the [unofficial "nightly" releases](https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation/actions) (actually, workflow runs)

Maybe the bug has been fixed on the `main` branch without anyone noticing? 👀

**Comment by cretoo** [comment]
@Rudxain Thanks i'll take a look at it. It seems like UADNG only list system apps and not preinstalled user apps.

**Comment by Rudxain** [comment]
> @Rudxain Thanks i'll take a look at it.

You're welcome! BTW, those releases are generally triggered manually, so even those might be "old" compared to `main`.

> preinstalled user apps.

Such as system apps that got updated? 👀

**Comment by cretoo** [comment]
> Such as system apps that got updated?

I don't think so, it's a brand new phone out of the box. I was able to see the mentioned apps (Netflix, facebook, etc.) listed under user apps on ADB AppControl which is another debloater, so i thought UADNG only list system apps and not user apps.

**Comment by Rudxain** [comment]
> only list system apps and not user apps.

Similar (old) report: 0x192/universal-android-debloater#795

It seems like an ADB bug. Or maybe the OEM has hidden those packs from ADB for some reason? It seems plausible, considering Samsung Galaxy "Deep Sleeping" apps are considered "Disabled" by the Android system

**Comment by AnonymousWP** [comment]
So I guess we can close this issue then considering both things are out of our control.

**Comment by Rudxain** [comment]
@cretoo run this cmd:
```sh
adb shell pm list packages -3
```
> [!note]
> for privacy, don't post the output

Does it print the missing packs?

If not, try this:
```sh
adb shell pm list packages `-a`
```
> [!warning]
> DO NOT POST THE FULL OUTPUT!

If any of those work, then we can fix it on our side, but it has to be a new toggle setting. The `-a` flag is very verbose, so it'll be annoying for some users.

I suspect the OEM has misleadingly marked as "user" some "system" packs, that's why `pm list packages -s` is so empty

**Comment by Rudxain** [comment]
huh, I thought `-a` was the default. This proves me wrong:
```sh
pm list packages -a | wc -l # 399
pm list packages | wc -l # 237
```

**Comment by Frigyes06** [comment]
Uad only lists system packages. Some phones have some apps as system vs third party. These apps are usually easily deletable from the device UI, however.

**Comment by Rudxain** [comment]
Related #607

**Comment by zalnaRs** [comment]
I have a Redmi Note 13 Pro 5G and the POCO X6 5G both are the same phones. I see all of them

**Comment by AnonymousWP** [comment]
Not sure what we should do with this issue now to be honest. As Frigyes said: you can easily uninstall/remove pre-installed user-apps easily.

**Comment by Rudxain** [comment]
IMO, we should simply show all packs (as an opt-in), not just `-s`. But this is far from a priority, unless there's a device model where at least 1 pack in `-3` cannot be disabled/uninstalled

**Comment by AnonymousWP** [comment]
Closing this issue because of Frigyes' and my conclusion. We can always re-open it later, but it's by design (you can just uninstall user-installed apps by yourself).

**Comment by Rudxain** [comment]
I agree, but

> uninstall user-installed apps

not all `-3` are manually-installed by the user 😕. Yes, they can be disabled by ADB and Settings-app, but what about the other features UAD-NG provides? (description, removal-type, backups, etc...)
